tests/components/element-editor/conditional-rendering/conditional-rendering.spec.tsx

```diff
@@ -0,0 +1,122 @@
+import { fireEvent, render, screen } from "@testing-library/react";
+import "jest-canvas-mock";
+import { ConditionalRendering } from "src/components/element-editor/conditional-rendering";
+import { EElementType } from "src/context-providers";
+import { ELEMENT_BUTTON_LABELS } from "src/data";
+import { TestHelper } from "src/util/test-helper";
+
+describe("ConditionalRendering", () => {
+    beforeEach(() => {
+        global.ResizeObserver = jest.fn().mockImplementation(() => ({
+            observe: jest.fn(),
+            unobserve: jest.fn(),
+            disconnect: jest.fn(),
+        }));
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+        jest.resetAllMocks();
+    });
+
+    it("should contain the multi entry component with the title, buttonLabel & children being passed into it", () => {
+        renderComponent({
+            builderContext: {
+                focusedElement: { element: MOCK_ELEMENT },
+                elements: MOCK_ELEMENTS,
+            },
+        });
+        expect(screen.getByText("Conditional Rendering")).toBeInTheDocument();
+        expect(screen.getByText("Add condition")).toBeInTheDocument();
+    });
+
+    it("should fire onAdd and render the condition-child component when the button is being clicked", () => {
+        renderComponent({
+            builderContext: {
+                focusedElement: { element: MOCK_ELEMENT },
+                elements: MOCK_ELEMENTS,
+            },
+        });
+        fireEvent.click(
+            screen.getByRole("button", {
+                name: "Add condition",
+            })
+        );
+
+        expect(screen.getByText("Select")).toBeInTheDocument();
+        expect(screen.getByText("Equals")).toBeInTheDocument();
+        expect(screen.getByPlaceholderText("Set value")).toBeInTheDocument();
+    });
+
+    it("should disable the add button when there is only one element in the elements list", () => {
+        renderComponent({
+            builderContext: {
+                focusedElement: { element: MOCK_ELEMENT },
+                elements: MOCK_ONE_ELEMENT,
```

Comment:
```suggestion
                elements: { [MOCK_ELEMENT.internalId]: MOCK_ELEMENT },
```
can delete `MOCK_ONE_ELEMENT`